internal/iapl/rbac.go

```diff
@@ -0,0 +1,222 @@
+	RoleBindingActionCreate RoleBindingAction = "rolebinding_create"
+	// RoleBindingActionUpdate is the action name to update a role binding
+	RoleBindingActionUpdate RoleBindingAction = "rolebinding_update"
+	// RoleBindingActionDelete is the action name to delete a role binding
+	RoleBindingActionDelete RoleBindingAction = "rolebinding_delete"
+	// RoleBindingActionGet is the action name to get a role binding
+	RoleBindingActionGet RoleBindingAction = "rolebinding_get"
+	// RoleBindingActionList is the action name to list role bindings
+	RoleBindingActionList RoleBindingAction = "rolebinding_list"
+)
+
+// ResourceRoleBindingV2 describes the relationships that will be created
+// for a resource to support role-binding V2
+type ResourceRoleBindingV2 struct {
+	// InheritPermissionsFrom is the list of resource types that can provide roles
+	// and grants to this resource
+	// Note that not all roles are available to all resources. This relationship is used to
+	// determine which roles are available to a resource.
+	// Before creating a role binding for a resource, one should check whether or
+	// not the role is available for the resource.
+	//
+	// Also see the RoleOwners field in the RBAC struct
+	InheritPermissionsFrom []string
+}
+
+/*
+RBAC represents a role-based access control policy.
+
+For example, consider the following spicedb schema:
+```zed
+
+	definition user {}
+	definition client {}
+
+	definition group {
+		relation member: user | client
+	}
+
+	definition organization {
+		relation parent: organization
+		relation member: user | client
+		relation member_role: role
+		relation grant: rolebinding
+
+		permissions rolebinding_list: grant->rolebinding_list
+		permissions rolebinding_create: grant->rolebinding_create
+		permissions rolebinding_delete: grant->rolebinding_delete
+	}
+
+	definition role {
+		relation owner: organization
+		relation view_organization: user:* | client:*
+		relation rolebinding_list_rel: user:* | client:*
+		relation rolebinding_create_rel: user:* | client:*
+		relation rolebinding_delete_rel: user:* | client:*
+	}
+
+	definition rolebinding {
+		relation role: role
+		relation subject: user | group#member
+		permission view_organization = subject & role->view_organization
+		permissions rolebinding_list: subject & role->rolebinding_list
+		permissions rolebinding_create: subject & role->rolebinding_create
+		permissions rolebinding_delete: subject & role->rolebinding_delete
+	}
+
+```
+in IAPL policy terms:
+- the RoleResource would be "{name: role, idprefix: someprefix}"
+- the RoleBindingResource would be "{name: rolebinding, idprefix: someprefix}",
+- the RoleRelationshipSubject would be `[user, client]`.
+- the RoleBindingSubjects would be `[{name: user}, {name: group, subjectrelation: member}]`.
+*/
+type RBAC struct {
+	// RoleResource is the name of the resource type that represents a role.
+	RoleResource RBACResourceDefinition
```

Comment:
Is `RBACResourceDefinition` here so we can create the resource type?

Reply:
Yes. I changed that to be not private because I realized at some point we need to create a policy for tests